carbon-design-system/carbon
# fix(type): ensure all tokens are configurable

Closes #12697 

A few type tokens were missed and not added to be `show`n through `@carbon/styles` (and therefore `@carbon/react`, too).

#### Changelog

**Changed**

- added all type tokens to `show` list in `@carbon/styles` to ensure they're configurable
- added `!default` to the `$tokens` map in the type module to ensure that it is configurable

#### Testing / Reviewing

- Pull down the PR, and modify `examples/vite/src/index.scss` to include the following:
```scss
@use '@carbon/react/scss/type' with (
  $body-compact-01: (
    font-size: 50rem,
    font-weight: 600,
    line-height: 1.28572,
    letter-spacing: 0.16px,
  )
);
@use '@carbon/react';

```
- Install and start the vite example, it should compile and show a change to type styling
<img width="776" alt="image" src="https://user-images.githubusercontent.com/3360588/210881373-85a83b96-911b-467e-9612-30531243d609.png">
